File: public/examples/integrations/mcp-servers/figma/get_pages_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()

USER_ID = "{arcade_user_id}"
TOOL_NAME = "Figma.GetPages"

auth_response = client.tools.authorize(tool_name=TOOL_NAME, user_id=USER_ID)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

client.auth.wait_for_completion(auth_response)

tool_input = {
    "file_key": "abc123xyz",  # Replace with your Figma file key
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/figma/get_pages_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade();

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "Figma.GetPages";

const authResponse = await client.tools.authorize({tool_name: TOOL_NAME, user_id: USER_ID});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

await client.auth.waitForCompletion(authResponse);

const toolInput = {
  file_key: "abc123xyz",  // Replace with your Figma file key
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));

